src/main/java/seedu/address/model/policy/Coverage.java

```diff
@@ -61,14 +61,22 @@ public Period getPeriod(String days, String months, String years) {
      * Returns true if a given string is a valid coverage declaration.
      */
     public static boolean isValidCoverage(String coverage) {
+        requireNonNull(coverage);
         ArrayList<String> coverageBreakDown = getCoverageBreakDown(coverage);
         String days = coverageBreakDown.get(dayIndex);
         String months = coverageBreakDown.get(monthIndex);
         String years = coverageBreakDown.get(yearIndex);
         boolean emptyOrMissingAllPrefixes = days.equals("0") && months.equals("0") && years.equals("0");
-        return !emptyOrMissingAllPrefixes
-                && days.matches(VALIDATION_REGEX) && months.matches(VALIDATION_REGEX)
+        boolean matchesRegex = days.matches(VALIDATION_REGEX) && months.matches(VALIDATION_REGEX)
                 && years.matches(VALIDATION_REGEX);
+        if (!emptyOrMissingAllPrefixes && matchesRegex) {
+            boolean isValidDay = Integer.parseInt(days) >= 0 && Integer.parseInt(days) <= 31;
+            boolean isValidMonth = Integer.parseInt(months) >= 0 && Integer.parseInt(months) <= 12;
+            boolean isValidYear = Integer.parseInt(years) >= 0;
+            return isValidDay && isValidMonth && isValidYear;
+        } else {
```

Comment:
Need to fix error message too.
![Screenshot 2019-11-05 at 10 01 10 AM](https://user-images.githubusercontent.com/40198156/68172652-4d24d880-ffb3-11e9-8c5c-de63b92e8464.png)